pkg/handler/collector/github/github.go

```diff
@@ -0,0 +1,140 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/guacsec/guac/pkg/handler/collector"
+	"github.com/guacsec/guac/pkg/handler/collector/file"
+	"github.com/guacsec/guac/pkg/handler/processor"
+	"golang.org/x/oauth2"
+)
+
+const (
+	CollectorGitHubDocument = "GitHubCollector"
+)
+
+type githubDocumentCollector struct {
+	url           string
+	dir           string
+	lastChecked   time.Time
+	poll          bool
+	interval      time.Duration
+	fileCollector collector.Collector
+}
+
+func NewGitHubDocumentCollector(ctx context.Context, url string, dir string, poll bool, interval time.Duration) *githubDocumentCollector {
+	fileCollector := file.NewFileCollector(ctx, dir, false, time.Second)
+
+	return &githubDocumentCollector{
+		url:           url,
+		dir:           dir,
+		poll:          poll,
+		interval:      interval,
+		fileCollector: fileCollector,
+	}
+}
+
+func (g *githubDocumentCollector) RetrieveArtifacts(ctx context.Context, docChannel chan<- *processor.Document) error {
+	// Replace with your own personal access token, export to use
+	token := os.Getenv("API_KEY")
```

Comment:
Shouldn't this be more specific to a github api key / PAT?

Reply:
@JonZeolla you're right it should. I'm going to make that more apparent as I push this scaffolding further. Thanks for the headsup!

Reply:
I would probably also pass this into the `githubDocumentCollector` and either call the environment variable, config file, etc. from what calls the githubDocumentCollector's constructor. I could also see falling back to looking for environment variable in the constructor if a token is not passed to it.